docs/advanced/migrating-buidler-plugin.md

```diff
@@ -0,0 +1,154 @@
+# Migrating a plugin from Buidler to hardhat
+
+This is a short guide explaining how to turn a Buidler plugin into a Hardhat one.
+
+If you want a complete example of a Hardhat plugin, take a look at [this repository](https://github.com/nomiclabs/buidler-ts-plugin-boilerplate/).
+
+## Updating its dependencies
+
+### Core package
+
+References to the `@nomiclabs/buidler` package should be replaced with the `hardhat` package in your `package.json`, and your `import`s or `require`s.
+
+For example, you would import the `extendEnvironment` function this way:
+
+```typescript
+import { extendEnvironment } from "hardhat/config";
+```
+
+### Plugins
+
+Similarly, references to Buidler plugins should be replaced with their corresponding hardhat plugins.
+For example, `@nomiclabs/buidler-ethers` would be `@nomiclabs/hardhat-ethers`.
+
+## Adapting your plugin's source code
+
+Replace all types or imported names that include `Buidler` with `Hardhat` in your plugin source code.
+
+For example, the `BuidlerRuntimeEnvironment` should be replaced with the `HardhatRuntimeEnvironment`. We suggest using `hre` instead of `bre` as its variable name.
+
+### Artifacts
+
+The `readArtifact` and `readArtifactSync` functions were moved to the `HardhatRuntimeEnvironment` so you must replace their uses like this:
+
+```js
+const tokenArtifact = await hre.artifacts.readArtifact("Token");
+```
+
+The artifact format is now supplemented with build information and debug artifacts in Hardhat which allows you to read things like contract symbols. See the [documentation](https://hardhat.org/docs/artifacts) for more information.
+
+## Updating your plugin's tests
+
+Apart from updating types and names, fixture projects need their `buidler.config.js` renamed to `hardhat.config.js`.
+
+### Changes needed to your test projects' config
+
+The compiler configuration is now expected in the `solidity` field instead of `solc`. Note that Hardhat projects allow multiple solidity versions in its compilation pipeline. For more information see its [documentation](https://hardhat.org/docs/compilation).
```

Comment:
```suggestion
The compiler configuration is now expected in the `solidity` field instead of `solc`. Note that Hardhat projects allow multiple solidity versions in its compilation pipeline. For more information see its [documentation](../guides/compile-contracts).
```